server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -410,4 +417,65 @@ private void assertIndexDirectoryDoesNotExist(String... indexNames) {
             }
         }
     }
+
+    public void testCacheFilesAreClosedAfterUse() throws Exception {
+        final int numReplicasIndex = randomIntBetween(1, 4);
+        final String indexName = "test-idx";
+        final String restoredIndexName = indexName + "-copy";
+        final String repoName = "test-repo";
+        final String snapshotName = "test-snap";
+        final String id = randomAlphaOfLength(5);
+        final Client client = client();
+
+        internalCluster().ensureAtLeastNumSearchAndDataNodes(numReplicasIndex + 1);
+        createIndex(indexName);
+        client().prepareIndex(indexName).setId(id).setSource("field", "test").get();
+        ensureGreen();
+        createRepositoryWithSettings(null, repoName);
+        takeSnapshot(client, snapshotName, repoName, indexName);
+        restoreSnapshotAndEnsureGreen(client, snapshotName, repoName);
+
+        // Search document to make the index fetch data from the remote snapshot to local storage
+        SearchResponse searchResponse = client().prepareSearch(restoredIndexName).setQuery(QueryBuilders.termQuery("field", "test")).get();
+        assertHitCount(searchResponse, 1);
+
+        // Get the primary shards for the given index
+        final ClusterState state = client().admin().cluster().prepareState().get().getState();
+        final Index index = state.metadata().index(restoredIndexName).getIndex();
+        final GroupShardsIterator<ShardIterator> shardIterators = state.getRoutingTable()
+            .activePrimaryShardsGrouped(new String[] { restoredIndexName }, false);
+
+        // The local cache files should be closed by deleting the restored index
+        deleteIndicesAndEnsureGreen(client, restoredIndexName);
+
+        logger.info("--> validate all the cache files are closed");
+        // Iterate all the primary shards to get all the possible path that contains local cache file
```

Comment:
@tlfeng With the changes within #6350, you can get easy access to the cache node path and check for files particularly within the cache instead of step walking through the `indices` directory. 
And this code will pass always since there will be no shards in this directory when #6350 is merged in. 
I think you would want to base your test off the changed files for the cache logic.

Reply:
@kotwanikunal Thank you! Since you wrote lots of code about the cache, the constructive idea is what I want to hear. I will change the test to match the latest code.